app/src/main/java/com/github/se/studybuddies/database/DatabaseConnection.kt

```diff
@@ -981,33 +974,42 @@ class DatabaseConnection : DbRepository {
         })
   }
 
-  override fun startDirectMessage(otherUID: String) {
+  override suspend fun startDirectMessage(otherUID: String): String {
     val currentUserUID = getCurrentUserUID()
+    val contactsViewModel = ContactsViewModel()
+    var contactID = ""
+    Log.d("MyPrint", "startDirectMessage called in db")
     checkForExistingChat(currentUserUID, otherUID) { chatExists, chatId ->
       if (chatExists) {
         Log.d("MyPrint", "startDirectMessage: chat already exists with ID: $chatId")
+        if (chatId != null) {
+          contactID = chatId
+        }
       } else {
         Log.d("MyPrint", "startDirectMessage: creating new chat")
         val newChatId = UUID.randomUUID().toString()
+        contactID = newChatId
         val memberPath = getPrivateChatMembersPath(newChatId)
         val members = mapOf(currentUserUID to true, otherUID to true)
         rtDb
             .getReference(memberPath)
             .updateChildren(members)
             .addOnSuccessListener {
               Log.d("DatabaseConnect", "startDirectMessage : Members successfully added!")
+              contactsViewModel.createContact(otherUID, contactID)
```

Comment:
A viewModel has nothing to do in the DatabaseConnection. 
You can call the DatabaseConnection from the viewModel, but not the other direction

Reply:
this is a temporary fix to a syncronization problem, thanks for pointing it out i could have forgot it

Reply:
i found a way to solve this while implementing the contact request, so you will see it solved in the next PR